### DISCUSSION TITLE
Backups made under same (latest) version cannot be restored after reinstalling the app

### DISCUSSION THREAD

**Discussion by Pophigist** [libre-tube/LibreTube#6375] [General]
### Steps to reproduce

v.0.24.0 - I create a full backup via the app
Unistall Libretube
Reinstall same version
Go to Backup & restore, choose Restore

### Expected behavior

Restoring part or all settings of the app should be offered and commence upon aproval

### Actual behavior

Cannot select the backup file for restoration, no matter where the backup file is located (SD card - root or subdirectory, - or root of the internal storage, or a directory within it).
As is the case for any unrelated file, the backup's name is visible in navigation but can't be selected.
BTW, if I create a backup file now it's just 4KB small as opposed to approx. 157KB prior to reinstalling.

### LibreTube version

0.24.0

### Android version

8.1

### Other details

In the last week or two the app has been plagued by inability to play or download via any server (even though I always use direct download of data from Youtube). Besides the above issue with restoration being unavailable upon reinstallation, my playlists, which have been backed up separately, could only be restored empty (lists` names only). And now "sign in to prove you are not a bot" alert is also shown instead of playback.
All in all, a lot of problems have been plaguing the previously near bug-free app since early August, in my experience.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
I can't reproduce being unable to selext a backup file - please share which Android ROM you use and the file manager you use to select the file. Also make sure the file name ends with .json.

> my playlists, which have been backed up separately, could only be restored empty
>
They can't be restored while Piped fails to extract video information, thus that's expected and will work again in case Piped is fixed one day.

**Comment by Pophigist** [comment]
Thank you for your prompt response.
Attached are the details re the phone (no modifications to the manufacturer's ROM, no rooting etc.; the file manager is something generic by Lenovo - version 4.03.1417_VER_32526591323981 -  that's not available on Google Play but has been bug-free since 2018). 
The backup files - I've made a few over time but never attempted recovery - are indeed json files. I can try restoring on a different phone if you think it'd help troubleshooting. Judging by file sizes if compared to a backup of a fresh install, the json files should contain the relevant info... 
[photo_2024-08-19_21-25-38](https://github.com/user-attachments/assets/43f27135-d117-4695-aaa5-cd7023db337e)

**Comment by Bnyro** [comment]
If you're able to, it would be great to test importing on an other device :+1:

**Comment by Pophigist** [comment]
Problem solved by finding a workaround.
Long story short, backups saved via the standard file manager of my phone are fine (as in restoring OK on another phone) yet cannot be opened via same file manager if I attempt restoring Libretube on that very phone.
Installing another file manager (fossify.org seemed safe enough to try) solved the problem: all backup files are available for Libretube via that app.

Thanks for your efforts!
Hope things get even smoother over time...
_Shalom from Israel_

**Comment by Bnyro** [comment]
Good to know :+1:

I've already heard from similar problems at other apps, so I suspect that some old pre-installed file managers are just kind of broken and don't handle the Anfroid file manager intent actions (asking Android for selecting a file, creating one, ...) probably, and using an old Android version probably causes thr file manager version to be old (and not compatible with new Android libraries) as well.
